lib/chai/core/assertions.js

```diff
@@ -235,55 +346,144 @@ module.exports = function (chai, _) {
   Assertion.addChainableMethod('a', an);
 
   /**
-   * ### .include(value)
+   * ### .include(val[, msg])
+   *
+   * When invoked as a method, and when the target is a string, `.include`
+   * asserts that the given string `val` is a substring within the target
+   * string.
+   *
+   *     expect('foobar').to.include('foo');
+   *
+   * When invoked as a method, and when the target is an array, `.include`
+   * asserts that the given `val` is a member of the target array. By default,
+   * strict (`===`) equality is used to compare members.
+   *
+   *     expect([1, 2, 3]).to.include(2);
+   *
+   * When invoked as a method, and when the target is an object, `.include`
+   * asserts that the given object `val`'s own enumerable properties are a
+   * subset of the target's enumerable and non-enumerable properties. By
+   * default, the target's own and inherited properties are included in the
+   * search. By default, strict equality is used to compare property values.
+   *
+   *     expect({a: 1, b: 2, c: 3}).to.include({a: 1, b: 2});
+   *
+   * Because the behavior of `.include` differs based on the target's type, it's
+   * important to always pair it with another assertion that verifies the
+   * target's type. Otherwise, `.include` will produce inaccurate results if a
+   * bug is later introduced in your code that changes the target's type. See
+   * the `.a` assertion for details on how to test a target's type.
+   *
+   *     expect([1, 2, 3]).to.be.an('array').that.includes(2);
+   *
+   * If the `deep` flag is set when the target is an array or object, deep
```

Comment:
💭 I wonder if it is worth adding headers to these, to break up the flow a bit and also aid in discerning the individual features. Right now it feels like they blend into one and other, which is not ideal.